fvm/evm/types/result.go

```diff
@@ -127,6 +130,54 @@ func (res *Result) VMErrorString() string {
 	return ""
 }
 
+// ErrorMsg returns the error message, if any VM or Validation error
+// both error would never happen at the same time
+// but if it happens the priority is by validation error
+func (res *Result) ErrorMsg() string {
+	errorMsg := ""
+	if res.VMError != nil {
+		errorMsg = res.VMError.Error()
+	}
+	if res.ValidationError != nil {
+		errorMsg = res.ValidationError.Error()
+	}
+	return errorMsg
+}
+
+// RLPEncodedLogs returns the rlp encoding of the logs
+func (res *Result) RLPEncodedLogs() ([]byte, error) {
+	var encodedLogs []byte
+	var err error
+	if len(res.Logs) > 0 {
+		encodedLogs, err = rlp.EncodeToBytes(res.Logs)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return encodedLogs, nil
+}
+
+// DeployedContractAddressString returns an string of the deployed address
+// it returns an empty string if the deployed address is nil
+func (res *Result) DeployedContractAddressString() string {
+	deployedAddress := ""
+	if res.DeployedContractAddress != nil {
+		deployedAddress = res.DeployedContractAddress.String()
+	}
+	return deployedAddress
+}
+
+// StateChangeChecksum constructs a checksum
+// based on the state change commitment on the result
+func (res *Result) StateChangeChecksum() [ChecksumLength]byte {
+	// the first 4 bytes of StateChangeCommitment is used as checksum
+	var checksum [ChecksumLength]byte
+	if len(res.StateChangeCommitment) >= ChecksumLength {
+		copy(checksum[:ChecksumLength], res.StateChangeCommitment[:ChecksumLength])
+	}
```

Comment:
better extract this into a public function to avoid re-implementing it elsewhere.

Reply:
updated to be a public function